validphys2/src/validphys/config.py

```diff
@@ -232,22 +232,9 @@ def produce_fitinputcontext(self, fit):
         _, theory = self.parse_from_("fit", "theory", write=False)
         thid = theory["theoryid"]
 
-        # new fits have dataset_inputs, old fits have experiments
-        data_key = "dataset_inputs"
-        try:
-            _, data_val = self.parse_from_("fit", data_key, write=False)
-        except ConfigError as e:
-            data_key = "experiments"
-            log.warning("old fit found, falling back to old behaviour")
-            # We need to make theoryid available if using experiments
-            try:
-                with self.set_context(ns=self._curr_ns.new_child({"theoryid": thid})):
-                    _, data_val = self.parse_from_("fit", data_key, write=False)
-            except ConfigError as inner_error:
-                raise e from inner_error
-        # now fill in a unique key "data_input" regardless of new or old fit
-        # for uniformity
-        data_input = self.produce_data_input(**{data_key: data_val})
+        data_input = self._parse_data_input_from_(
+            "fit", {"theoryid": thid}
```

Comment:
@Zaharid  do you know why ` {"theoryid": thid}` is passed here? It seems unnecessary to me.

Reply:
It may have had something to do with how `experiments` work(ed). But seems unneeded for getting the right `dataset_inputs`.